drivers/can/can_nxp_s32_canxl.c

```diff
@@ -0,0 +1,977 @@
+		timing->prop_seg, timing->phase_seg1, timing->phase_seg2,
+		timing->prescaler, timing->sjw);
+
+	canxl_timing->propSeg = timing->prop_seg - 1U;
+	canxl_timing->phaseSeg1 = timing->phase_seg1 - 1U;
+	canxl_timing->phaseSeg2 = timing->phase_seg2 - 1U;
+	canxl_timing->preDivider = timing->prescaler - 1U;
+	canxl_timing->rJumpwidth = timing->sjw - 1U;
+}
+
+static int can_nxp_s32_set_timing(const struct device *dev,
+			const struct can_timing *timing)
+{
+	const struct can_nxp_s32_config *config = dev->config;
+	struct can_nxp_s32_data *data = dev->data;
+	Canexcel_Ip_TimeSegmentType can_time_segment = {0};
+
+	if (data->started) {
+		return -EBUSY;
+	}
+
+	LOG_DBG("Setting CAN bitrate %d:", config->bitrate);
+	nxp_s32_zcan_timming_to_canxl_timming(timing, &can_time_segment);
+
+	/* Set timing for CAN instance*/
+	CanXL_SetBaudRate(config->base_sic, &can_time_segment);
+
+	return 0;
+}
+
+#ifdef CONFIG_CAN_FD_MODE
+static int can_nxp_s32_set_timing_data(const struct device *dev,
+			const struct can_timing *timing_data)
+{
+	const struct can_nxp_s32_config *config = dev->config;
+	struct can_nxp_s32_data *data = dev->data;
+	Canexcel_Ip_TimeSegmentType can_fd_time_segment = {0};
+
+	if (data->started) {
+		return -EBUSY;
+	}
+
+	LOG_DBG("Setting CAN-FD bitrate %d:", config->bitrate_data);
+	nxp_s32_zcan_timming_to_canxl_timming(timing_data, &can_fd_time_segment);
+
+	/* Set timing for CAN FD instance*/
+	CanXL_SetFDBaudRate(config->base_sic, &can_fd_time_segment);
+
+	return 0;
+}
+#endif
+
+void can_int_error_isr(const struct device *dev)
+{
+	const struct can_nxp_s32_config *config = dev->config;
+
+	Canexcel_Ip_ErrIRQHandler(config->instance);
+}
+
+void can_int_rx_tx_data_isr(const struct device *dev)
+{
+	const struct can_nxp_s32_config *config = dev->config;
+
+	Canexcel_Ip_RxTxIRQHandler(config->instance);
+}
+
+static void can_nxp_s32_err_callback(const struct device *dev,
+					Canexcel_Ip_EventType eventType,
+					uint32 u32SysStatus,
+					const Canexcel_Ip_StateType *canexcelState)
+{
+	const struct can_nxp_s32_config *config = dev->config;
+	struct can_nxp_s32_data *data = dev->data;
+	enum can_state state;
+	struct can_bus_err_cnt err_cnt;
+	void *cb_data = data->state_change_cb_data;
+	can_tx_callback_t function;
+	int alloc;
+	void *arg;
+
+	switch (eventType) {
+	case CANEXCEL_EVENT_TX_WARNING:
+		LOG_WRN("Tx Warning (error 0x%x)", u32SysStatus);
+		break;
+	case CANEXCEL_EVENT_RX_WARNING:
+		LOG_WRN("Rx Warning (error 0x%x)", u32SysStatus);
+		break;
+	case CANEXCEL_EVENT_BUSOFF:
+		LOG_ERR("Bus Off (error 0x%x)", u32SysStatus);
+		break;
+	case CANEXCEL_EVENT_ERROR:
+		LOG_ERR("Error Format Frames (error 0x%x)", u32SysStatus);
+		break;
+	case CANEXCEL_EVENT_ERROR_FD:
+		LOG_ERR("Error Data Phase (error 0x%x)", u32SysStatus);
+		break;
+	case CANEXCEL_EVENT_PASSIVE:
+		LOG_ERR("Error Passive (error 0x%x)", u32SysStatus);
+		break;
+	default:
+		break;
+	}
```

Comment:
Either remove these or change them to `LOG_DBG()`. This will spam the log in perfectly valid network situations.

Reply:
I changed to LOG_DBG